Interface usuario e fluxo execucao

DraNefario opened the pull request:
Este PR visa implementar a parte 3 do sistema especialista para recomendação de trilha acadêmcia.

Funcionalidades implementadas

faz_perguntas -> Percorre todas as perguntas
perguntar -> Realiza as perguntas individuais (é chamada pela funçao faz_perguntas)
validar_resposta -> Garante que o usuario digite apenas s/n
fluxo principal na ordem ((iniciar -> perguntar -> calcular -> exibir)

Cenários testados:

Teste de entrada "s" e "n"
Responder "s" ou "n" para as perguntas interativas usando perguntar/2.
Verificar se a resposta foi armazenada corretamente com resposta(Id, s) ou resposta(Id, n).


Teste de repetição do questionário
Confirmar que em caso de resposta inválida não ocorre repetição de todo o questionário.
Apenas a pergunta atual é reexecutada, mantendo as respostas anteriores armazenadas.


Teste de fluxo completo
Utilizar faz_perguntas para percorrer todas as perguntas sequencialmente.
Verificar se todas as respostas válidas foram capturadas e armazenadas.

Ambiente de entrada
Testes no SWISH, onde a entrada via botão "Send" sempre captura corretamente.
Testes em terminal local com diferentes métodos (read_line_to_string/2 e read/1) para garantir que o método read/1 funcione melhor em certos ambientes.

DraNefario committed:
adicionei funçao que percorre todas as perguntas

DraNefario committed:
funcao "perguntar" chamada por faz_perguntas para realizar perguntas individuais

DraNefario committed:
funcao "validar_resposta" para garantir que a resposta seja apenas s/n

DraNefario committed:
fluxo principal

Guilherme006 commented:
@DraNefario o fluxo ficou claro, mas penso que há 4 pontos que podem ser ajustados:

- **Loop de perguntas:** o padrão com ; faz_perguntas pode gerar loop. Use forall/2 para percorrer todas as pergunta/3;
- **Entrada do usuário:** read/1 exige termo Prolog (s.). Trocar por read_line_to_string/2 + normalização, aceitando s/n;
- **Respostas duplicadas:** se errar e repetir, fica mais de um resposta(Id,_). Antes de salvar, faça retractall/1;
- **Limpar estado no início:** garantir retractall(resposta(_,_)) em iniciar/0.

DraNefario committed:
Update base_conhecimento.pl

DraNefario committed:
foram aplicadas as sugestoes do Guilherme006

Loop de perguntas: o padrão com ; faz_perguntas pode gerar loop. Use forall/2 para percorrer todas as pergunta/3;
Entrada do usuário: read/1 exige termo Prolog (s.). Trocar por read_line_to_string/2 + normalização, aceitando s/n;
Respostas duplicadas: se errar e repetir, fica mais de um resposta(Id,_). Antes de salvar, faça retractall/1;
Limpar estado no início: garantir retractall(resposta(,)) em iniciar/0.

DraNefario committed:
Merge pull request #5 from danltargz/main

pull request main para a minha branch

danltargz reviewed (approved):
LGTM!
